### ISSUE TITLE
pdns crashes when database insert fails to postgres database (lowercase constraint)

### ISSUE THREAD

**Issue by saltsa** [PowerDNS/pdns#3583]
In the current pdns setup, when the AXFR transfer contains names which has uppercase letters the following happens:
```
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 Transaction started for 'nnn.fi.'
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 Unable to feed record during incoming AXFR of 'nnn.fi.': GSQLBackend unable to feed record: Fatal error during query: insert into records (content,ttl,prio,type,domain_id,disabled,name,ordername,auth,change_date) values ($1,$2,$3,$4,$5,$6,$7,$8,$9,null): ERROR:  new row for relation "records" violates check constraint "c_lowercase_name"
Mar 16 04:58:37 ns3 pdns_server: DETAIL:  Failing row contains (6802, 5, 0A.client.nnn.fi, A, 192.168.0.10, 300, 0, null, f, null, t).
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 Aborting possible open transaction for domain 'nnn.fi.' AXFR
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 Got a signal 11, attempting to print trace:
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(+0x18195b) [0x7fde2958495b]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /lib64/libc.so.6(+0x35670) [0x7fde27948670]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/pgsql-9.5/lib/libpq.so.5(PQclear+0x69) [0x7fde272c0f49]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/lib64/pdns/libgpgsqlbackend.so(_ZN15SPgSQLStatementD0Ev+0x5d) [0x7fde274f4f5d]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(_ZN11GSQLBackend14freeStatementsEv+0x199) [0x7fde294967d9]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/lib64/pdns/libgpgsqlbackend.so(_ZN13gPgSQLBackendD0Ev+0x34) [0x7fde274f1d24]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(_ZN12UeberBackend7cleanupEv+0x184) [0x7fde295d9f84]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(_ZN12UeberBackendD2Ev+0x2b) [0x7fde295da03b]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(_ZN17CommunicatorClass4suckERK7DNSNameRKSs+0x5fc) [0x7fde295b211c]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(_ZN17CommunicatorClass19retrievalLoopThreadEv+0x136) [0x7fde294aded6]
Mar 16 04:58:37 ns3 pdns_server: Mar 16 10:58:37 /usr/sbin/pdns_server(+0xaea69) [0x7fde294b1a69]
Mar 16 04:58:38 ns3 pdns_server: Mar 16 10:58:37 /lib64/libpthread.so.0(+0x7dc5) [0x7fde27cdbdc5]
Mar 16 04:58:38 ns3 pdns_server: Mar 16 10:58:37 /lib64/libc.so.6(clone+0x6d) [0x7fde27a0928d]
```

**Comment by saltsa** [comment]
The version tested was:
Version     : 4.0.0
Release     : 0.1.alpha2.1pdns.el7

**Comment by pieterlexis** [comment]
The crash should be fixed with #3612, but this should never habben again because of #3621
